### PR TITLE
More bug fixes

### PR DESCRIPTION
**Pull request by DinoV**

### CODE REVIEW

File: Pyjion/absint.cpp [hunk 1]
```diff
@@ -132,6 +132,15 @@ bool AbstractInterpreter::preprocess() {
                     }
                 }
                 break;
+            case LOAD_GLOBAL:
+            {
+                auto name = PyUnicode_AsUTF8(PyTuple_GetItem(m_code->co_names, oparg));
+                if (!strcmp(name, "vars") || !strcmp(name, "dir")) {
```
Comment (brettcannon) [inline Pyjion/absint.cpp]: Probably should add a comment that we are aware this fails if either `vars()` or `dirs()` is bound locally, but that's simply a limitation people will have to live with and avoid.

File: Pyjion/absint.cpp [hunk 2]
```diff
@@ -200,16 +209,27 @@ bool AbstractInterpreter::interpret() {
...
-                    lastState[lastState.stack_size() - 2] = tmp;
+                    // ROT_TWO currently assumes we have native ints on the stack,
+                    // so we force escape here...  We should fix that up in the future
+                    // and use pop_no_escape
+                    auto tmp = lastState.pop();
+                    auto second = lastState.pop();
+                    lastState.push(tmp);
+                    lastState.push(second);
+                    
+                    //auto tmp = lastState[lastState.stack_size() - 1];
```
Comment (brettcannon) [inline Pyjion/absint.cpp]: Test code?

File: Pyjion/absint.cpp [hunk 3]
```diff
@@ -200,16 +209,27 @@ bool AbstractInterpreter::interpret() {
                 case NOP: break;
                 case ROT_TWO:
                 {
-                    auto tmp = lastState[lastState.stack_size() - 1];
-                    lastState[lastState.stack_size() - 1] = lastState[lastState.stack_size() - 2];
-                    lastState[lastState.stack_size() - 2] = tmp;
+                    // ROT_TWO currently assumes we have native ints on the stack,
+                    // so we force escape here...  We should fix that up in the future
```
Comment (brettcannon) [inline Pyjion/absint.cpp]: Probably should explicitly add a TODO here (or add an issue).

File: Pyjion/absint.cpp [hunk 4]
```diff
@@ -200,16 +209,27 @@ bool AbstractInterpreter::interpret() {
...
+                    //lastState[lastState.stack_size() - 2] = tmp;
                     break;
                 }
                 case ROT_THREE:
                 {
-                    auto top = lastState.pop_no_escape();
-                    auto second = lastState.pop_no_escape();
-                    auto third = lastState.pop_no_escape();
+                    // ROT_THREE currently assumes we have native ints on the stack,
+                    // so we force escape here...  We should fix that up in the future
```
Comment (brettcannon) [inline Pyjion/absint.cpp]: Explicit `TODO` or open an issue.

File: Pyjion/absint.cpp (hunk 3 above)
Comment (DinoV) [inline Pyjion/absint.cpp]: Opened issue #88 for both of them

**Comment by brettcannon** [comment]
I didn't see any tests for the vars()/dirs() case nor the bytes multiplication (only str multiplication).

File: Pyjion/absint.cpp (hunk 2 above)
Comment (DinoV) [inline Pyjion/absint.cpp]: The commented out code is the ideal code (which is correct), we just need to update the code generator for what's now issue #88.  Once we do that this code can come back.

Comment (brettcannon) [inline Pyjion/absint.cpp]: Ah, OK. Then a quick comment along those lines would be great.

**Comment by DinoV** [comment]
test_builtin covers the vars/dir case and I think that's sufficient.  I'll add a bytes test case.

**Comment by brettcannon** [comment]
Just address the LOAD_GLOBAL comment and then this can be merged.

**Pull request merged**